mantidproject/mantid
# Prevent GUI thread locking using MWRunFiles

This prevents a very long running file search from blocking the GUI event thread. It also gets what was the `FindFilesThread` under unit test.

Instead of the blocking approach we were using before, MWRunFiles now using a thread pool to create a worker that searches for files. Importantly, if a new search is initiated we will not block and wait for the thread to finish.

**To test:**
 - Make sure that data archive search is turned on.
 - The original bug that this was reported for is described in the issue. Please check that this no longer occurs.
 - Also check that opening and quickly closing the GUI does not cause it to hang or crash.

This widget has wide use in Mantid. The interfaces that currently use this widget are:
 - The Load Dialog
 - The Fit Dialog
 - ISIS Engineering Diffraction Interface
 - ISIS Muon Interface
 - ISIS Indirect Interfaces
 - ISIS SANS Interface
 - Anywhere that uses the `DataProcessor` widget (ISIS Reflectometry)

Fixes #20855 

**Release Notes** 

See [here](https://github.com/mantidproject/mantid/pull/20954/commits/30bc63610b8a50f51bedd6666d349806d6a14845).

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
